app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/settings/accountsecurity/loginapproval/LoginApprovalViewModel.kt

```diff
@@ -207,11 +207,17 @@ class LoginApprovalViewModel @Inject constructor(
     }
 
     private fun closeScreen() {
-        if (state.specialCircumstance?.shouldFinishWhenComplete == true) {
-            sendEvent(LoginApprovalEvent.ExitApp)
+        sendEventBasedOnSpecialCircumstance()
```

Comment:
Would it make sense to just call `sendEventBasedOnSpecialCircumstances` directly instead of `closeScreen`?

Reply:
Great call! Guess my love of Matryoshka dolls got the better of me :)